### PR TITLE
FISH-6737 Upgrade Tool fails when domain1 has been removed and there are more than 1 domains

### PR DESCRIPTION
**Pull request by JamesHillyard**
By setting the domain name, when the initDomain method gets the domain name for the DomainDirs constructor, it bypasses the need to look for 'domain1' which has been hardcoded. Normally this would cause the command to be executed against that specific domain, but in the case of upgrade tool it will upgrade all domains anyway, so using a specific domain here doesn't change anything. This also maintains the full validation of the domains dir.

Tested with 0 domains, 1 domain, 2 domains (inc domain 1), 2 domains (no domain 1) - All scenarios worked

Pending approval, I will update this to 1.6, release to nexus, tag the release, and increment to 1.7-SNAPSHOT

### CODE REVIEW

**Review by Pandrex247 — CHANGES_REQUESTED** [review]
I found a failing case:

If the default domain dir has no entries or the default entries, you get a failure if you use a custom domain dir where the names of the domains don't match.

So, for example, in default domain dir have the usual domain1 and production, and then in a custom domains dir have two custom domains (e.g. adomain and bdomain).

When you use the upgrade server command and specify your custom domain dir you get an error stating that domain1 doesn't exist.

You hit essentially the same problem if the default domain dir is empty - it's not looking at your defined domain dir.

TLDR: like in the "original" `initDomain` method you need to account for the domaindir param.

**Comment by JamesHillyard** [comment]
Tested: 
- Default domaindir, domain1 & production
- Default domaindir, production only
- Default domaindir, domain1 only
- Default domaindir, _empty_

These specify a custom domaindir with --domaindir, leaving the default dir in various states
- Custom domaindir (adomain, bdomain), default domaindir (domain1, production)
- Custom domaindir (adomain, bdomain), default domaindir (domain1)
- Custom domaindir (adomain, bdomain), default domaindir (production)
- Custom domaindir (adomain, bdomain), default domaindir _(empty)_
- Custom domaindir (domain1), default domaindir _(empty)_
- Custom domaindir (domain1), default domaindir (domain1)
- Custom domaindir _(empty)_, default domaindir (domain1, production)
- Custom domaindir _(empty)_, default domaindir _(empty)_

File: src/main/java/fish/payara/extras/upgrade/BaseUpgradeCommand.java [hunk 1]
```diff
@@ -129,6 +131,28 @@ protected void validate() throws CommandException {
...
+     */
+    @Override
+    protected void initDomain() throws CommandException {
+        try {
+             File defaultDomainsDir = getDefaultDomainsDir();
+             //defaultDomainsDir can't be null here as getDefaultDomainsDir() will throw an IOException in that scenario
+             if (defaultDomainsDir.listFiles().length == 0) {
+                 throw new CommandException("There are no domains in " + defaultDomainsDir);
+             }
+             setDomainName(defaultDomainsDir.listFiles()[0].getName());
```
[the remark this answers is not in the document]
Comment (JamesHillyard) [reply src/main/java/fish/payara/extras/upgrade/BaseUpgradeCommand.java]: Added

**Review by Pandrex247 — APPROVED** [review]
